app/src/main/java/swati4star/createpdf/fragment/ExtractTextFragment.java

```diff
@@ -0,0 +1,217 @@
+                    mTextView.setText(fileName);
+                    mTextView.setVisibility(View.VISIBLE);
+                    extractText.setEnabled(true);
+                    mMorphButtonUtility.morphToSquare(extractText, mMorphButtonUtility.integer());
+                }
+                break;
+        }
+        super.onActivityResult(requestCode, resultCode, data);
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (grantResults.length < 1)
+            return;
+        switch (requestCode) {
+            case PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT: {
+                if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    mPermissionGranted = true;
+                    openExtractText();
+                    showSnackbar(mActivity, R.string.snackbar_permissions_given);
+                } else
+                    showSnackbar(mActivity, R.string.snackbar_insufficient_permissions);
+            }
+        }
+    }
+
+    @OnClick(R.id.extract_text)
+    public void openExtractText() {
+        if (!mPermissionGranted) {
+            getRuntimePermissions();
+            return;
+        }
+        new MaterialDialog.Builder(mActivity)
+                .title(R.string.creating_pdf)
+                .content(R.string.enter_file_name)
+                .input(getString(R.string.example), null, (dialog, input) -> {
+                    if (StringUtils.isEmpty(input)) {
+                        showSnackbar(mActivity, R.string.snackbar_name_not_blank);
+                    } else {
+                        final String inputName = input.toString();
+                        if (!mFileUtils.isFileExist(inputName + ".txt")) {
+                            extractTextfromPdf(inputName);
+                        } else {
+                            MaterialDialog.Builder builder = createOverwriteDialog(mActivity);
+                            builder.onPositive((dialog12, which) -> extractTextfromPdf(inputName))
+                                    .onNegative((dialog1, which) -> openExtractText())
+                                    .show();
+                        }
+                    }
+                })
+                .show();
+    }
+
+    private void extractTextfromPdf(String inputName) {
+        String mStorePath = mSharedPreferences.getString(STORAGE_LOCATION,
+                getDefaultStorageLocation());
+        String mPath = mStorePath + inputName + textExtension;
+        try {
+            String parsedText = "";
+            PdfReader reader = new PdfReader(mRealPath);
+            int n = reader.getNumberOfPages();
+            for (int i = 0; i < n; i++) {
+                parsedText = parsedText + PdfTextExtractor.getTextFromPage(reader, i + 1)
+                        .trim() + "\n"; //Extracting the content from the different pages
```

Comment:
How about using `StringBuilder` instead of a string?

A good resource to learn more: https://stackoverflow.com/questions/2971315/string-stringbuffer-and-stringbuilder

Reply:
Yeah I know about `StringBuilder` will be updating that. Thanks.